### ISSUE TITLE
node-http-handler: probable regression for requestTimeout

### ISSUE THREAD

**Issue by raymondfeng** [aws/aws-sdk-js-v3#4618]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

The following commit https://github.com/aws/aws-sdk-js-v3/commit/86a6046af170784a9858f101e84cf3d97df84a21 changes the behavior of node-http-handler, used by SQS client:

```
ReceiveMessageCommandInput https://sqs.us-west-1.amazonaws.com/220623082201/collabland-jobs-test: {
   MessageAttributeNames: [ '*' ],
   WaitTimeSeconds: 20,
   MaxNumberOfMessages: 10,
   VisibilityTimeout: 300,
   QueueUrl: 'https://sqs.us-west-1.amazonaws.com/220623082201/collabland-jobs-test'
 } +2ms

 Fail to receive/delete messages from https://sqs.us-west-1.amazonaws.com/220623082201/collabland-jobs-test: Error [TimeoutError]: Connection timed out after 5000 ms
     at ClientRequest.<anonymous> (/Users/rfeng/Projects/collabland/collabland-monorepo/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:91:38)
     at Object.onceWrapper (node:events:627:28)
     at ClientRequest.emit (node:events:513:28)
     at TLSSocket.emitRequestTimeout (node:_http_client:838:9)
     at Object.onceWrapper (node:events:627:28)
     at TLSSocket.emit (node:events:525:35)
     at Socket._onTimeout (node:net:562:8)
     at listOnTimeout (node:internal/timers:564:17)
     at process.processTimers (node:internal/timers:507:7) {
   '$metadata': { attempts: 1, totalRetryDelay: 0 }
 } +5s
```

### SDK version number

"@aws-sdk/node-http-handler": "^3.306.0"

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

18

### Reproduction Steps

- Use AWS SDK to receive messages from a queue that does not have any messages

### Observed Behavior

- An error is thrown:

```
[TimeoutError]: Connection timed out after 5000 ms
     at ClientRequest.<anonymous> (/Users/rfeng/Projects/collabland/collabland-monorepo/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:91:38)
     at Object.onceWrapper (node:events:627:28)
     at ClientRequest.emit (node:events:513:28)
     at TLSSocket.emitRequestTimeout (node:_http_client:838:9)
     at Object.onceWrapper (node:events:627:28)
     at TLSSocket.emit (node:events:525:35)
     at Socket._onTimeout (node:net:562:8)
     at listOnTimeout (node:internal/timers:564:17)
     at process.processTimers (node:internal/timers:507:7) {
   '$metadata': { attempts: 1, totalRetryDelay: 0 }
 }
```

### Expected Behavior

It should return empty message array. The connectionTimeout/requestTimeout should be used to control `connection`, not the http response.

### Possible Solution

Workaround - increase requestTimeout to 30s

**Comment by yenfryherrerafeliz** [comment]
Hi @raymondfeng, sorry to hear about your issues. I feel this is the expected behavior because, the requestTimeout parameter is designed to control how long a request should wait for it to complete, and in this case, since the time set for waiting for a new message is longer than the timeout set for the request to complete it indeed will throw a timeout error. However, I will bring this into discussion with the team to analyze this better.

Thanks!

**Comment by raymondfeng** [comment]
I don't think it's by design, at least it's a breaking change. The docs says `requestTimeout` is for connection, not reading messages. Throws an error in polling messages is not desired.

**Comment by trivikr** [comment]
The fix to undeprecate `connectionTimeout` and reintroduce functionality was pushed in PR https://github.com/aws/aws-sdk-js-v3/pull/4669
It'll be released with v3.322.0 around 1pm Pacific either on Thu 04/27 or Fri 04/28

In the meantime, please use <= v3.295.0 as a workaround.

**Comment by trivikr** [comment]
The fix was released in https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.321.1
